flink-libraries/flink-table/src/test/java/org/apache/flink/api/java/table/test/StringExpressionsITCase.java

```diff
@@ -40,46 +40,6 @@ public StringExpressionsITCase(TestExecutionMode mode) {
 		super(mode);
 	}
 
-	@Test(expected = CodeGenException.class)
```

Comment:
No, the `ScalarFunctionsTest` does not test end-to-end. I don't think this is necessary for scalar functions, because they don't affect optimization/plan translation. When we will have more functions (e.g. 40 of them), we would have 40 end-to-end tests just implementing a MapFunction and evaluating an expression. That's why I implemented the `ExpressionEvaluator`.